users/clients/api/transport.go

```diff
@@ -301,7 +301,6 @@ func decodePasswordResetRequest(_ context.Context, r *http.Request) (interface{}
 		return nil, errors.Wrap(errors.ErrMalformedEntity, err)
 	}
 
-	req.Host = r.Header.Get("Referer")
```

Comment:
why is this removed

Reply:
It kept on resulting in the `req.host` being blank. I pass the `host` in the body.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer

Reply:
I know what the referer is. In this case I kept on getting the error `missing host` in Mainflux-Users service. That's why I removed it from here.

Reply:
On the request you can pass it as a header
```go
req.Header.Add("Referer", "<referer url>")
```

Reply:
This would take in the request url of the sdk `processRequest ` method, not the request of my UI, hence it would not work. I pass the host as an sdk configuration. Now not unless we add it here in the header https://github.com/mainflux/mainflux/blob/57d47fea66193b762f6ac68db757d5fc26d7bc82/pkg/sdk/go/sdk.go#L1005

Reply:
You can change process Request to accept headers

Reply:
This would change `processRequest` just for one function. I only need Referer for this function only. And I can pass it in the body together with the email and decode it here: https://github.com/mainflux/mainflux/blob/57d47fea66193b762f6ac68db757d5fc26d7bc82/users/clients/api/transport.go#L300, that way it will only be in that method, as I don't see the need of adding a `referer` header in the processRequest as other methods don't need it. Is there any particular reason you really need this line: https://github.com/mainflux/mainflux/blob/57d47fea66193b762f6ac68db757d5fc26d7bc82/users/clients/api/transport.go#L304

Reply:
I think it's ok to extend processRequest even if it is only used in one method. Other methods can benefit in the future